sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/XmlExternalEntityShouldNotBeParsedTest.cs

```diff
@@ -91,9 +91,13 @@ public void XmlExternalEntityShouldNotBeParsed_AlwaysSafe(NetFrameworkVersion ve
         [DataRow(NetFrameworkVersion.Between4And451, @"TestCases\XmlExternalEntityShouldNotBeParsed_XmlReader_Net4.cs")]
         [DataRow(NetFrameworkVersion.After452, @"TestCases\XmlExternalEntityShouldNotBeParsed_XmlReader_Net452.cs")]
         [DataRow(NetFrameworkVersion.Unknown, @"TestCases\XmlExternalEntityShouldNotBeParsed_XmlReader_Net452.cs")]
+        [DataRow(NetFrameworkVersion.Probably35, @"TestCases\XmlExternalEntityShouldNotBeParsed_XPathDocument_Net35.cs")]
+        [DataRow(NetFrameworkVersion.Between4And451, @"TestCases\XmlExternalEntityShouldNotBeParsed_XPathDocument_Net4.cs")]
+        [DataRow(NetFrameworkVersion.After452, @"TestCases\XmlExternalEntityShouldNotBeParsed_XPathDocument_Net452.cs")]
+        [DataRow(NetFrameworkVersion.Unknown, @"TestCases\XmlExternalEntityShouldNotBeParsed_XPathDocument_Net452.cs")]
         [DataTestMethod]
         [TestCategory("Rule")]
-        public void XmlExternalEntityShouldNotBeParsed_XmlReader(NetFrameworkVersion version, string testFilePath)
+        public void XmlExternalEntityShouldNotBeParsed_XmlReaderAndXPathDocument(NetFrameworkVersion version, string testFilePath)
```

Comment:
why not splitting in a different test? if there's a regression in the XmlReader impl, there's no need for the XPathDocument to fail

Reply:
Indeed that would be confusing. I wanted to reuse the code but I can also extract a method to achieve this :)